common/src/com/intellij/plugins/haxe/util/HaxeCommonCompilerUtil.java

```diff
@@ -95,155 +100,229 @@ public static boolean compile(final CompilationContext context) {
       context.log(HaxeCommonBundle.message("module.0.is.excluded.from.compilation", context.getModuleName()));
       return true;
     }
+
+    if (!verifyProjectSettings(context)) {
+      return false;
+    }
+
+    final String workingPath = calculateWorkingPath(context);
+    if (! workingPath.isEmpty()) {
+      context.setErrorRoot(workingPath);
+    }
+    final File workingDirectory = new File(FileUtil.toSystemDependentName(workingPath));
+    if (!workingDirectory.exists()) {
+      if (!workingDirectory.mkdirs()) {
+        context.errorHandler(HaxeCommonBundle.message("output.path.not.found", workingPath));
+        return false;
+      }
+    }
+
+    final List<List<String>> commandLines = generateCommandLines(context);
+
+    final BooleanValueHolder hasErrors = new BooleanValueHolder(false);
+    try {
+      for (List<String> commandLine : commandLines) {
+
+        // XXX: Would it be useful to show the working directory in the output window??
+        // If so, then just call notifyTextAvailable() before startNotify() below.
+
+        // Show the command line in the output window.
+        // TODO: Make a checkbox in the SDK configuration window to enable/disable showing the command line.
+        String commandLineString = String.join(" ", commandLine);
+
+        final BaseOSProcessHandler handler = new HaxeCompilerProcessHandler(
+          context,
+          HaxeSdkUtilBase.createProcessBuilder(commandLine, workingDirectory, context.getHaxeSdkData()).start(),
+          commandLineString,
+          Charset.defaultCharset()
+        );
+
+        handler.addProcessListener(new ProcessAdapter() {
+          @Override
+          public void processTerminated(ProcessEvent event) {
+            int exitcode = event.getExitCode();
+            hasErrors.setValue(exitcode != 0);
+            if (exitcode < 0) {
+              context.infoHandler(HaxeCommonBundle.message("negative.error.code.message"));
+            }
+
+            super.processTerminated(event);
+          }
+        });
+
+        handler.startNotify();
+        handler.waitFor();
+      }
+    }
+    catch (IOException e) {
+      context.errorHandler(HaxeCommonBundle.message("process.threw.exception", e.getMessage()));
+      hasErrors.setValue(true);
+      return false;
+    }
+
+    return !hasErrors.getValue();
+  }
+
+  private static boolean verifyProjectSettings(CompilationContext context) {
+    final HaxeModuleSettingsBase settings = context.getModuleSettings();
     final String mainClass = context.getCompilationClass();
     final String fileName = context.getOutputFileName();
+    boolean requiresHaxelib = false;
+
 
     if (settings.isUseUserPropertiesToBuild()) {
-      if (mainClass == null || mainClass.length() == 0) {
-        context.errorHandler(HaxeCommonBundle.message("no.main.class.for.module", context.getModuleName()));
+
+      if ( !verifyNonEmptyString(mainClass, context, "no.main.class.for.module")
```

Comment:
How do you think - is it be better to show user all problems at once, not one by one?

Reply:
That's an interesting point.  I hadn't thought about it because I was just repackaging the original code.  Rather than short-circuit, I'll put out the error messages and remember that there was an error, then return appropriately at the end.

Reply:
That would be great!